djangosaml2idp/idp.py

```diff
@@ -1,53 +1,68 @@
-import copy
-
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
+from django.http import HttpRequest
 from django.utils.translation import gettext as _
 from saml2.config import IdPConfig
 from saml2.metadata import entity_descriptor
 from saml2.server import Server
+from typing import Callable, Dict, Optional, Union
+
+from .conf import get_callable, get_config
 
 
 class IDP:
     """ Access point for the IDP Server instance
     """
-    _server_instance: Server = None
+    _server_instances: Dict[str, Server] = {}
```

Comment:
There are now several instances cached : the keys of this dict are the `entityid`s of the metadata. So it is expected the IDP conf is persistent for one `entityid` (see the load method)